cmd/transport/base-controller.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2023 The KubeStellar Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/spf13/pflag"
+
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog/v2"
+
+	transportoptions "github.com/kubestellar/kubestellar/cmd/transport/options"
+	edgeclientset "github.com/kubestellar/kubestellar/pkg/generated/clientset/versioned"
+	edgeinformers "github.com/kubestellar/kubestellar/pkg/generated/informers/externalversions"
+	"github.com/kubestellar/kubestellar/pkg/transport"
+)
+
+const (
+	defaultResyncPeriod = time.Duration(0)
+)
+
+func Run(transportImplementation transport.Transport) {
+	logger := klog.Background().WithName(transport.ControllerName)
+	ctx := klog.NewContext(context.Background(), logger)
+
+	options := transportoptions.NewTransportOptions()
+	fs := pflag.NewFlagSet(transport.ControllerName, pflag.ExitOnError)
+	klog.InitFlags(flag.CommandLine)
+	fs.AddGoFlagSet(flag.CommandLine)
+	options.AddFlags(fs)
+	fs.Parse(os.Args[1:])
+
+	fs.VisitAll(func(flg *pflag.Flag) {
+		logger.Info(fmt.Sprintf("Command line flag '%s' - value '%s'", flg.Name, flg.Value)) // log all arguments
```

Comment:
Why not use the defined structure of Logger::Info --- the keysAndValues? Something like

```
logger.Info("Command line flag", flg.Name, flg.Value)
```

or

```
logger.Info("Command line flag", "name", flg.Name, "value", flg.Value)
```

Reply:
this is how I wrote it initially but it had some linter warnings.
(at least it used to have linter warnings when I worked on transport against release 0.15).
if linter configuration has changed and this doesn't create warnings I'm happy to do this change.

Reply:
The first of the two I suggested causes linter warnings; the second does not (in my experience in this project).

Reply:
done